libs/langchain/langchain/chains/base.py

```diff
@@ -127,6 +128,7 @@ def invoke(
         tags = config.get("tags")
         metadata = config.get("metadata")
         run_name = config.get("run_name") or self.get_name()
+        run_id = config.get("run_id")
```

Comment:
We should do the same in `ainvoke` !

Reply:
Before approving, we need:
1. add support in `ainvoke`, `stream`, `astream`
2. Add unit tests for each method to confirm they work

Reply:
@hinthornw 
Thank you. 
I added the run_id in the `ainvoke`, `stream` and `astream` here https://github.com/langchain-ai/langchain/pull/20889/commits/2386d31382282dd1e28615ea3cf8ce5b6d1a7cf3
I also took a look at unit tests, for example: 
- https://github.com/langchain-ai/langchain/blob/master/libs/langchain/tests/unit_tests/chains/test_base.py
- https://github.com/langchain-ai/langchain/blob/master/libs/langchain/tests/unit_tests/agents/test_agent.py
- https://github.com/langchain-ai/langchain/blob/master/libs/langchain/tests/unit_tests/agents/test_agent_iterator.py
But I don't see the relative test about config attributes like `run_name`. Could you guide me on the test?
Our project needs this feature now, so I would be very happy if you could help us soon.

Reply:
@hinthornw 
https://github.com/langchain-ai/langchain/pull/20889#issuecomment-2084768994
Could you read this comment? This problem is blocking our process so I would appreciate your help.

Reply:
Sorry about this! notifications have been inundated from other things. Will review now